sender/sender.go

```diff
@@ -227,3 +230,166 @@ func Push2InfluxdbSendQueue(items []*cmodel.MetaData) {
 		}
 	}
 }
+
+// Push network quality metric pkt to the queue for RPC
+func Push2NqmRpcSendQueue(items []*cmodel.MetaData) {
+	for _, item := range items {
+		if item.Metric == "nqm-metrics" {
+			//log.Println("Push2NqmRpcSendQueue: found nqm pkt.", item)
+			nqmitem, err := convert2NqmRpcItem(item)
+			if err != nil {
+				log.Println("NqmRpc converting error:", err)
+				continue
+			}
+			isSuccess := NqmRpcQueue.PushFront(nqmitem)
+
+			if !isSuccess {
+				proc.SendToNqmRpcDropCnt.Incr()
+			}
+		}
+	}
+}
+
+func convert2NqmRpcItem(d *cmodel.MetaData) (*nqmRpcItem, error) {
+	var t nqmRpcItem
+	agent, err := convert2NqmEndpoint(d, "agent")
+	if err != nil {
+		return &t, err
+	}
+	target, err := convert2NqmEndpoint(d, "target")
+	if err != nil {
+		return &t, err
+	}
+	metrics, err := convert2NqmMetrics(d)
+	if err != nil {
+		return &t, err
+	}
+
+	t = nqmRpcItem{
+		Timestamp: d.Timestamp,
+		Agent:     *agent,
+		Target:    *target,
+		Metrics:   *metrics,
+	}
+
+	return &t, nil
+}
+
+func convert2NqmEndpoint(d *cmodel.MetaData, endtype string) (*nqmEndpoint, error) {
+	t := nqmEndpoint{
+		Id:         -1,
+		IspId:      -1,
+		ProvinceId: -1,
+		CityId:     -1,
+		NameTagId:  -1,
+	}
+
+	var ii int64
+	var err error
+	if v, present := d.Tags[endtype+"-id"]; present {
+		ii, err = strconv.ParseInt(v, 10, 32)
+		t.Id = int(ii)
+		if err != nil {
+			return &t, err
+		}
+	}
```

Comment:
1. The `if err != nil` statement should be put after `ii, err = strconv.ParseInt(v, 10, 32)` immediately.
1. While **error is not nil**, use `nil` as return value would be better
1. If you like more information for error context, try using `fmt.Errorf()` to generate detail information.

For example:
```
v, err = someFunc()
if err != nil {
    return nil, fmt.Errorf("Your error context: %v", err)
}
```